### ISSUE TITLE
Building packages  with meta.yaml including the text '.hg' anywhere

### ISSUE THREAD

**Issue by icobo** [conda/conda-build#1488]
Hello!

The first of all, congratulations for the conda project!! It's very useful for our projects, maintaining the reproducibility of our workflows. 

We are having problems on building recipes with the text **hg** included in any part of the **meta.yaml** file text.
For checking it (and discarding that the dot character was causing the error), we copy two recipes from conda and build them.
- The **recip_bsgenome** recipe has character dots in the _meta.yaml_ and it works perfectly.
- The **recip_bsgenome_with_gh** has the sub string **.hg** inside and it _breaks_ with this error:

```
*** more dependencies output before here ***
 zlib:                              1.2.8-3                   
updating index in: /home/icobo/miniconda2/conda-bld/linux-64
updating index in: /home/icobo/miniconda2/conda-bld/noarch

The following NEW packages will be INSTALLED:

    mercurial:  3.9.2-py27_0 
    pip:        8.1.2-py27_0 
    python:     2.7.12-1     
    setuptools: 27.2.0-py27_0
    sqlite:     3.13.0-0     
    wheel:      0.29.0-py27_0

Source cache directory is: /home/icobo/miniconda2/conda-bld/src_cache
Found source in cache: BSgenome.Hsapiens.UCSC.hg19_1.4.0.tar.gz
Extracting download
Package: bioconductor-bsgenome.hsapiens.ucsc.hg19-1.4.0-r3.3.1_0
source tree in: /home/icobo/miniconda2/conda-bld/recip_bsgenome_with_words_1477569611046/work/BSgenome.Hsapiens.UCSC.hg19
+ source /home/icobo/miniconda2/bin/activate /home/icobo/miniconda2/conda-bld/recip_bsgenome_with_words_1477569611046/_b_env_
+ mv DESCRIPTION DESCRIPTION.old
+ grep -v '^Priority: ' DESCRIPTION.old
+ /home/icobo/miniconda2/conda-bld/recip_bsgenome_with_words_1477569611046/_b_env_/bin/R CMD INSTALL --build .
/home/icobo/miniconda2/conda-bld/recip_bsgenome_with_words_1477569611046/work/BSgenome.Hsapiens.UCSC.hg19/conda_build.sh: línea 10: /home/icobo/miniconda2/conda-bld/recip_bsgenome_with_words_1477569611046/_b_env_/bin/R: No existe el archivo o el directorio
Command failed: /bin/bash -x -e /home/icobo/miniconda2/conda-bld/recip_bsgenome_with_words_1477569611046/work/BSgenome.Hsapiens.UCSC.hg19/conda_build.sh
```

We have the suspicion that the error in the code is in the file [conda_build/metadata.py](https://github.com/conda/conda-build/blob/120791c3387848e89fbf187df264c938cbc3de4f/conda_build/metadata.py#L871), in the line **817**,   in the function **uses_vcs_in_build**.

Thank you very much for all!!.

[recip_bsgenome_with_gh.tar.gz](https://github.com/conda/conda-build/files/555665/recip_bsgenome_with_gh.tar.gz)
[recip_bsgenome.tar.gz](https://github.com/conda/conda-build/files/555666/recip_bsgenome.tar.gz)

**Comment by msarahan** [comment]
Thanks for the report.  This is a bug from an overly general regular expression that we try to use to ensure that people have appropriate version control tools installed at the time that a package is built.  I'll fix it for the next bug fix release.

**Comment by msarahan** [comment]
fixed by #1498 - closing.
